scripts/setup.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/sh
+# Don't use set -e to avoid terminal closing when sourced with . ./scripts/setup.sh
+# Instead, check return codes explicitly and report errors
+
+# Check if rad command is available
+if ! command -v rad >/dev/null 2>&1; then
+
+  echo "⛓️‍💥 This script requires the Radicle toolchain to be installed."
+  echo "  💡 Please install it from https://radicle.xyz/download and try again."
```

Comment:
Speaking for myself, I will not be able to install Radicle this way (it will be a declared dependency in `flake.nix`, which is how it ends up on my system).

You've mentioned that a skill-gated barrier to entry is acceptable (and perhaps even desirable) for certain kinds of participation, so: why not assume that `flake.nix` is comprehensive and require that `nix develop` be the first step for bootstrapping a local development environment?

Reply:
If you have Nix installed, you can run `nix develop` and it will drop you into a shell environment with all the build tools this project needs set up and ready to go (including Node.js, Rust toolchain and on Linux even the web browser and webdriver).

I like the script, since it seems like setting things up is pretty involved :shrug: